### PR TITLE
Add child stream handling to best practices

### PR DESCRIPTION
**Pull request by dmosorast**
This is a part of a lot of Singer taps, and if done right will make handling child streams a lot easier. We've wanted to include something like this for awhile, so I'm putting up this PR to start talking about how best to convey this guidance.

### CODE REVIEW

**Comment by asaf-erlich** [comment]
Great stuff overall. I'm glad you're going through these and improving upon them. The only feedback I have is to link to an example for each case showing how to do the right thing in code. It would be nice if all of them are the same tap too, but they don't have to be. I'm thinking tap-trello or tap-adroll are two recent and good examples written by us. People are going to tend to copy and paste and then modify it for their use case so the more times we can point to a specific example the better.

File: docs/BEST_PRACTICES.md [hunk 1]
```diff
@@ -143,3 +143,53 @@ install_requires=[
...
+     parent object and only making HTTP requests as necessary.
+   - [Example
+   tap-shopify](https://github.com/singer-io/tap-shopify/blob/v1.2.9/tap_shopify/streams/order_refunds.py#L28-L31)
+2. **Signpost State Keeping** - Set a maximum bookmark value before the
+   sync begins and store bookmarks only up to that value. For large data
+   sets, records are being updated as the tap is requesting all of the
+   data. As the tap's extraction intersects with the data changing in real
+   time, records can be missed, but setting a "signpost" bookmark as a
+   maximum value to be stored can help with many race conditions and
+   sorting issues.
```
Comment (aaronsteers) [inline docs/BEST_PRACTICES.md:179]: @dmosorast - Thanks for creating this content! I found the second sentence in this section a little hard to follow personally, specifically why/how records get missed. I have been struggling to document something similar myself so I took a stab below. It's hard to explain concisely but I think we're getting close.

```suggestion
   data. As the tap's extraction reads records that may be changing in real
   time, setting a "signpost" bookmark as a
   maximum value to be stored can help with many race conditions and
   sorting issues. This signpost prevents the bookmark from advancing to
   a point where subsequent executions of the tap would miss records modified
   within the batch interval.
```

Comment (dmosorast) [reply docs/BEST_PRACTICES.md:179]: Yup, this does read better. Thanks for the suggestion!

**Review by asaf-erlich — APPROVED** [review]
LGTM
